compiler/qsc_linter/src/lints/hir.rs

```diff
@@ -1,17 +1,106 @@
-use qsc_hir::hir::Lit;
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use qsc_hir::{
+    hir::{CallableDecl, CallableKind, Expr, ExprKind, SpecBody, SpecDecl, Stmt, StmtKind},
+    ty::Ty,
+    visit::{self, Visitor},
+};
 
 use crate::linter::hir::declare_hir_lints;
 
 use super::lint;
 
 declare_hir_lints! {
-    (Placeholder, LintLevel::Allow, "this a placeholder", "remove after addding the first HIR lint"),
+    (NeedlessOperation, LintLevel::Warn, "unnecessary operation declaration", "convert to function")
```

Comment:
Let's make the error message a bit more user-friendly. Here's my attempt:

```suggestion
    (NeedlessOperation, LintLevel::Warn, "operation does not contain any quantum operations", "this callable can be declared as a function instead")
```

Reply:
You're right. VS Code does support clickable error codes and we've started to add help documentation for some of them. We could do this for lint warnings as well. I'll see if I can set up a page like that.

![image](https://github.com/microsoft/qsharp/assets/16928427/0d6d526d-b272-49dd-8da8-130ccde30c96)

https://github.com/microsoft/qsharp/wiki/QIR#runtime-capabilities-check-errors